src/jdk.incubator.jextract/share/classes/jdk/internal/jextract/impl/TypeMaker.java

```diff
@@ -119,12 +182,16 @@ Type makeType(jdk.internal.clang.Type t) {
             }
             case Enum:
             case Record: {
-                return Type.declared((Declaration.Scoped)treeMaker.createTree(t.getDeclarationCursor()));
+                if (treeMaker == null) {
```

Comment:
Do we get a crash if we go here inside macro eval? Why the special casing - I mean I get that we probably don't care, but trying to understand.

Reply:
We get a NPE since treeMaker is null. If we simply plug-in a TreeMaker, then we get crash as TU in Macro is not usable.

I am thinking we could provide a inquiry-only TreeMaker of the precompile TU for macro case, then we can get the correct type.